pulsar-client/src/main/java/org/apache/pulsar/client/impl/TableViewImpl.java

```diff
@@ -304,7 +313,9 @@ private CompletableFuture<Map<String, TopicMessageId>> getLastMessageIds(Reader<
         return reader.getLastMessageIdsAsync().thenApply(lastMessageIds -> {
             Map<String, TopicMessageId> maxMessageIds = new ConcurrentHashMap<>();
             lastMessageIds.forEach(topicMessageId -> {
-                maxMessageIds.put(topicMessageId.getOwnerTopic(), topicMessageId);
+                if (((MessageIdAdv) topicMessageId).getEntryId() >= 0) {
```

Comment:
I think this is counter-intuitive that the the tableview needs to handle such invalid messageId in the downstream logic. Don't we need to add this logic in the upstream? For example, we could skip this message either in broker or consumer if the downstream shouldn't see such invalid messageIds?

Reply:
> we could skip this message either in broker or consumer if the downstream shouldn't see such invalid messageIds?

This message id actually has the same meaning with `MessageId.Earliest`, which does not represent the position of a message. Instead, it means all messages' ids should be greater than this message id, i.e. `x.compareTo(y)` will return false if `x` is such a message id and `y` is an id of a message.

For example, given a consumer that subscribes 2 topics ("tp1" with 2 partitions and "tp2" with 1 partition), the `getLastMessageIds` method will return a list

```
[0] TopicMessageId("tp1", {ledgerId: -1, entryId: -1, partition: 0})
[1] TopicMessageId("tp1", {ledgerId: -1, entryId: -1, partition: 1})
[2] TopicMessageId("tp2", {ledgerId: -1, entryId: -1, partition: 0})
```

Then the caller will know these two topics are subscribed by the current consumer. If you're going to skip these two message ids, the client side won't be able to see the subscribed topics from the result of `getLastMessageIds`. This would be a breaking change that could affect the downstream applications.

IMO, it would be better to add a method `isEarliest()` or `isLatest()` to `MessageId` (but it's beyond the scope of this PR). Actually `getLastMessageIds` does not return `MessageId.earliest`, whose partition index is -1. So I have to just check if `entryid` is negative here. You can also see the similar check internally in `ConsumerImpl#hasMessageAvailableAsync`.

Reply:
>  the tableview needs to handle such invalid messageId

1. When the message id represents "earliest", the future of this topic should be completed immediately.
2. Otherwise, the future of this topic will be completed after the reader reads the message whose message is greater than the message id.

It's intuitive because if the topic is empty, no messages will be read.

Reply:
I changed the method name and added comments for it now